src/sync.h

```diff
@@ -129,7 +127,25 @@ class LOCKABLE AnnotatedMixin : public PARENT
 using RecursiveMutex = AnnotatedMixin<std::recursive_mutex>;
 
 /** Wrapped mutex: supports waiting but not recursive locking */
-typedef AnnotatedMixin<std::mutex> Mutex;
+using Mutex = AnnotatedMixin<std::mutex>;
+
+/** Different type to mark Mutex at global scope
+ *
+ * Thread safety analysis can't handle negative assertions about mutexes
+ * with global scope well, so mark them with a separate type, and
+ * eventually move all the mutexes into classes so they are not globally
+ * visible.
+ *
+ * See: https://github.com/bitcoin/bitcoin/pull/20272#issuecomment-720755781
+ */
+class GlobalMutex : public Mutex { };
+
+#define AssertLockHeld(cs) AssertLockHeldInternal(#cs, __FILE__, __LINE__, &cs)
+
+inline void AssertLockNotHeldInline(const char* pszName, const char* pszFile, int nLine, Mutex* cs) EXCLUSIVE_LOCKS_REQUIRED(!cs) { AssertLockNotHeldInternal(pszName, pszFile, nLine, cs); }
```

Comment:
style: `AssertLockNotHeldInline()` is newly added function, so should abide to the naming convention for its arguments:

```suggestion
inline void AssertLockNotHeldInline(const char* mutex_name, const char* file_name, int line, Mutex* mutex) EXCLUSIVE_LOCKS_REQUIRED(!cs) { AssertLockNotHeldInternal(mutex_name, file_name, line, mutex); }
```